app/helpers/portal_helper.rb

```diff
@@ -8,4 +8,19 @@ def generate_portal_bg(portal_color, theme)
     bg_image = theme == 'dark' ? 'grid_dark.svg' : 'grid.svg'
     "background: url(/assets/images/hc/#{bg_image}) #{generate_portal_bg_color(portal_color, theme)}"
   end
+
+  def language_name(locale)
+    language_map = YAML.load_file(Rails.root.join('config/languages/language_map.yml'))
```

Comment:
This loads the yaml file with every function call, this could be in-efficient. 

You can declare a module variable at the top level

```rb
module PortalHelper
    @@language_map = nil
  
    # rest of the code
end
```

and update this method

```rb
def language_name(locale)
  @@language_map ||= YAML.load_file(Rails.root.join('config/languages/language_map.yml'))
  @@language_map[locale] || locale
end
```

This should effectively memoize this, as far as my understanding goes, this should persist across network requests in production. Not sure if this is a good pattern or not. CC: @sojan-official @pranavrajs

Reply:
@scmmishra Any idea 
<img width="1363" alt="image" src="https://github.com/chatwoot/chatwoot/assets/64252451/7b6db17f-76cb-4338-ad91-170eb8e7c1a3">

Reply:
🥲 @sojan-official @pranavrajs need your Ruby skills here.

@iamsivin feel free to revert that for now.

Reply:
Reverted it